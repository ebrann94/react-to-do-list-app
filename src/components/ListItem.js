import React from 'react';

const ListItem = (props) => {
    const className = props.completed ? 'completed list-item' : 'list-item';
    
    return (
        <div 
            className={className} 
            onClick={(e) => {
                if (e.target.id !== 'completed-checkbox') props.handleCompleteItem(props.id)
            }}
        >   
            <p>{props.text}</p>
            <div>
                <input 
                    type="checkbox" 
                    onChange={() => props.handleCompleteItem(props.id)} 
                    checked={props.completed}
                    className="list-item__checkbox"
                    id="completed-checkbox"
                />
                <button onClick={() => props.handleRemoveOne(props.id)}>Delete</button>
            </div>
        </div>
    );
}

export default ListItem;