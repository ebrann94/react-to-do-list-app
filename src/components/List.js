import React from 'react';
import ListItem from './ListItem';
import { TransitionGroup, CSSTransition } from 'react-transition-group';

const List = (props) => {
    return (
        <div className="items-container">
            <TransitionGroup className="todo-list">  
                {
                    props.items.map((item) => {
                        return (
                            <CSSTransition
                                classNames="fade"
                                timeout={200}
                                key={item.id}
                            >
                                <div className="list-item-wrapper">
                                    <ListItem 
                                        {...item}
                                        handleCompleteItem={props.handleCompleteItem}
                                        handleRemoveOne={props.handleRemoveOne}
                                    />
                                </div>
                            </CSSTransition>
                        );
                    })
                }
            </TransitionGroup>
        </div>
    );
    
}

export default List;