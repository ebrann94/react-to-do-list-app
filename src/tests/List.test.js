import React from 'react';
import { shallow } from 'enzyme';

import List from '../components/List';
import ListItem from '../components/ListItem';

describe('List component tests', () => {
    it('renders without crashing with required props', () => {
        const props = {
            items: [{
                id: '1234',
                text: 'Example text',
                completed: false
            }],
            handleCompleteItem: jest.fn(),
            handleRemoveOne: jest.fn()
        }
        const wrapper = shallow(<List {...props}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(ListItem)).toHaveLength(props.items.length);
    });
});