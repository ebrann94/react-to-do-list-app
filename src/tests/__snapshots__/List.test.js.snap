// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component tests renders without crashing with required props 1`] = `ShallowWrapper {}`;
